src/app/static/src/tests/components/runReport/runReport.test.js

```diff
@@ -133,11 +142,88 @@ describe("runReport", () => {
+        const wrapper = shallowMount(RunReport, {
+            propsData: {
+                metadata: {git_supported: true, instances_supported: false},
+                initialGitBranches
+            }
+        });
+
+        await Vue.nextTick();
+
+        expect(wrapper.find("#git-refresh-btn").exists()).toBe(true);
+        const button = wrapper.find("#git-refresh-btn");
+        expect(mockAxios.history.get.length).toBe(2);
+
+        expect(wrapper.find("#git-branch-form-group").exists()).toBe(true);
+        const options = wrapper.findAll("#git-branch-form-group select option");
+
+        expect(options.length).toBe(2);
+        expect(options.at(0).text()).toBe("master");
+        expect(options.at(0).attributes().value).toBe("master");
+        expect(options.at(1).text()).toBe("dev");
+        expect(options.at(1).attributes().value).toBe("dev");
+        
+        expect(wrapper.vm.$data.gitRefreshing).toBe(false);
+        expect(button.attributes("disabled")).toBeUndefined();
+        expect(button.text()).toBe("Refresh git");
+
+        button.trigger("click")
+        expect(wrapper.vm.$data.gitRefreshing).toBe(true);
+        await Vue.nextTick();
+        expect(button.attributes("disabled")).toBe("disabled");
+        expect(button.text()).toBe("Fetching...");
+        
+        setTimeout(() => {
+            expect(mockAxios.history.get.length).toBe(6);
+            expect(wrapper.vm.$data.gitRefreshing).toBe(false);
+            expect(button.attributes("disabled")).toBeUndefined();
+            expect(button.text()).toBe("Refresh git");
+
+            expect(options.length).toBe(2);
+            expect(options.at(0).text()).toBe("master2");
+            expect(options.at(0).attributes().value).toBe("master2");
+            expect(options.at(1).text()).toBe("dev2");
+            expect(options.at(1).attributes().value).toBe("dev2");
+            done();
+        })
+    });
+
+    it("show error message if error refreshing git", async (done) => {
+        mockAxios.onGet('http://app/git/fetch/')
+            .reply(500, "TEST ERROR");
+
+        const wrapper = shallowMount(RunReport, {
+            propsData: {
+                metadata: {git_supported: true, instances_supported: false},
+                initialGitBranches
+            }
+        });
+        expect(wrapper.find("#git-refresh-btn").exists()).toBe(true);
+        const button = wrapper.find("#git-refresh-btn");
+        button.trigger("click")
+        expect(wrapper.vm.$data.gitRefreshing).toBe(true);
+        await Vue.nextTick();
+        expect(button.attributes("disabled")).toBe("disabled");
+
+        setTimeout(() => {
+            expect(mockAxios.history.get.length).toBe(5);
+            expect(wrapper.vm.$data.gitRefreshing).toBe(false);
+            expect(button.attributes("disabled")).toBeUndefined();
+            // expect(wrapper.vm.$data.error).toBe("TEST ERROR");
```

Comment:
I've been unable to get the error messages to display as expected. I think it has to do with another failed request (likely the commits request on `changedBranch()`) being called after the failed fetch request, and overwriting the error data. I tried getting the logic to stop at the failed fetch request by adding a success condition in the then statement but that has not work:
```
api.get('/git/fetch/')
                    .then(({data}) => {
                        if (data.status === "success"){
                          this.gitRefreshing = false
                          this.gitBranches = data.data.map(branch => branch.name)
                        }
                    })
```

Reply:
OK, there's 3 things you need to fix here. 
1. You need to set up axios to give a response for the runnable reports request - this is actually happening in getWrapper - which you're not using - but it could equally, I think, move to `beforeEach` where other requests are mocked
2. You need to give the wrapper time to do all its setting up and fetching runnable reports before you test the git fetch button. So everything from clicking the button onwards needs to be wrapped in an outer `setTimeout`. See `clicking run button sends run request and displays status on success` for an example. You should really do this for the success case too. At the moment, the error is the error for the runnable reports request, which is happening *after* the git fetch call.
3. Need to match "TEST ERROR" against `vm.$data.error.response.data`